File: 剑指offer/day16-排序/45. 把数组排成最小的数.py
# 输入一个非负整数数组，把数组里所有数字拼接起来排成一个数，打印能拼接出的所有数字中最小的一个。

# 示例 1:
# 输入: [10,2]
# 输出: "102"

# 示例 2:
# 输入: [3,30,34,5,9]
# 输出: "3033459"


def minNumber(nums):
  def mySort(a, b):
    if a + b > b + a:
      return True
    else:
      return False

  nums = list(map(str, nums))
  nums.sort()

  # 冒泡排序
  for i in range(len(nums)):
    for j in range(len(nums) - i - 1):
      if mySort(nums[j], nums[j + 1]):
        nums[j], nums[j + 1] = nums[j + 1], nums[j]
  return nums


print(minNumber([10, 2]))
print(minNumber([3, 30, 34, 5, 9]))

File: 剑指offer/45. 把数组排成最小的数.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var minNumber = function (nums) {
  nums.sort()
  function mid(a, b) {
    return a + b > b + a ? true : false
  }

  for (let i = 0; i < nums.length - 1; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (mid(String(nums[i]), String(nums[j]))) {
        const temp = nums[i]
        nums[i] = nums[j]
        nums[j] = temp
      }
    }
  }
  return nums.join("")
};


console.log(minNumber([10, 2]));
console.log(minNumber([3, 30, 34, 5, 9]));